lib/SILOptimizer/Mandatory/FlowIsolation.cpp

```diff
@@ -565,6 +566,9 @@ void AnalysisInfo::analyze(const SILArgument *selfParam) {
     Operand *operand = worklist.pop_back_val();
     SILInstruction *user = operand->getUser();
 
+    if (!visited.insert(user))
```

Comment:
I know you are making a small improvement here, but IIRC the general pattern that we use is we guard the wordlist append with the set insert check rather than put the set insert check at the top of the list.

That being said, feel free to do that in a follow on PR.

Reply:
I created a helper `OperandWorklist` class. It's cleaner and also moves the insert check to the append